internal/lint/subscribe.go

```diff
@@ -12,35 +12,56 @@ import (
 
 // SubscriberInput for the shared test.
 type SubscriberInput struct {
-	// Subject should return a subject to be tested, with given topic/messages seeded.
+	// Subject should be set to a subject to be tested.
+	// It can optionally prepare (pre-create) or seed the topic at this point (the same topic will be used for Seed-ing later).
 	// Caller should handle all the subject teardown/cleanup.
-	// Topics are guaranteed to be prefixed with "bps-unittest-".
 	Subject func(topic string, messages []bps.SubMessage) bps.Subscriber
+
+	// Seed can seed given topic/messages for Subject unless already seeded.
+	// Topics are guaranteed to be prefixed with "bps-unittest-".
+	Seed func(topic string, messages []bps.SubMessage)
```

Comment:
should that return an `error`?

Reply:
Better not, leave that to caller (it's running in BeforeEach, so fine to use Expects in caller).